org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java

```diff
@@ -286,11 +288,21 @@ private void populateDataFields(List<Either<Command, CodeAction>> codeActions) {
 		IProblemLocationCore[] locations = new IProblemLocationCore[diagnostics.size()];
 		for (int i = 0; i < diagnostics.size(); i++) {
 			Diagnostic diagnostic = diagnostics.get(i);
-			int problemId = getProblemId(diagnostic);
 			int start = DiagnosticsHelper.getStartOffset(unit, diagnostic.getRange());
 			int end = DiagnosticsHelper.getEndOffset(unit, diagnostic.getRange());
 			boolean isError = diagnostic.getSeverity() == DiagnosticSeverity.Error;
-			locations[i] = new ProblemLocationCore(start, end - start, problemId, new String[0], isError, IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER);
+			int problemId = getProblemId(diagnostic);
+			String[] arguments;
+			if (diagnostic.getData() instanceof JsonArray) {
+				final JsonArray data = (JsonArray) diagnostic.getData();
+				arguments = new String[data.size()];
+				for (int j = 0; j < arguments.length; j++) {
+					arguments[j] = data.get(j).getAsString();
+				}
+			} else {
+				arguments = new String[0];
+			}
+			locations[i] = new ProblemLocationCore(start, end - start, problemId, arguments, isError, IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER);
```

Comment:
Just the following change for slightly improved readability. It should be the same behaviour.

```diff
diff --git a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java
index fd60ab29..f46e27be 100644
--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandler.java
@@ -63,6 +63,7 @@ import org.eclipse.lsp4j.WorkspaceEdit;
 import org.eclipse.lsp4j.jsonrpc.messages.Either;
 
 import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
 
 public class CodeActionHandler {
 	public static final ResponseStore<Either<ChangeCorrectionProposal, CodeActionProposal>> codeActionStore
@@ -292,17 +293,14 @@ public class CodeActionHandler {
 			int end = DiagnosticsHelper.getEndOffset(unit, diagnostic.getRange());
 			boolean isError = diagnostic.getSeverity() == DiagnosticSeverity.Error;
 			int problemId = getProblemId(diagnostic);
-			String[] arguments;
+			List<String> arguments = new ArrayList<>();
 			if (diagnostic.getData() instanceof JsonArray) {
 				final JsonArray data = (JsonArray) diagnostic.getData();
-				arguments = new String[data.size()];
-				for (int j = 0; j < arguments.length; j++) {
-					arguments[j] = data.get(j).getAsString();
+				for (JsonElement e : data) {
+					arguments.add(e.getAsString());
 				}
-			} else {
-				arguments = new String[0];
 			}
-			locations[i] = new ProblemLocationCore(start, end - start, problemId, arguments, isError, IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER);
+			locations[i] = new ProblemLocationCore(start, end - start, problemId, arguments.toArray(new String[0]), isError, IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER);
 		}
 		return locations;
 	}
```

Reply:
Fixed.